### PR TITLE
update pull request template

### PR DESCRIPTION
**Pull request by joeniquette**
## Description
We are looking to enhance our PR process for all repo's. Please suggest edits to make this the PR template as efficient as possible while also ensuring each PR meets a minimum standard of quality. This should increase the overall confidence of each code change being merged into main.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/49831

## Testing done
Template update, testing not required

## Screenshots
Template update, Screenshot not required

## Acceptance criteria
- [ ] Approval from each of the CoP leads
- [ ] Feedback integrated from all comments
- [ ] Approval from Platform Lead Engineer
- [ ] Approval from Platform Team Leaders

### CODE REVIEW

**Review by it-harrison — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bramleyjl) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: I'm a little unclear on this, is this a reference to updating automated/unit tests and the behaviors that they are testing? Or is this about further integration testing?

**Review by bramleyjl — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (AGarcia-Clarity) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Keep the inclusion of this link department-of-veterans-affairs/va.gov-team#0000. All tickets created in Zenhub will be on the va.gov-team repository

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (AGarcia-Clarity) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: - Remove and replace with a table for before & after with mobile & desktop
- This field is mandatory for component work and UI changes (non-component work should NOT have screenshots).

| | Before | After |
| --- | --- | --- |
| Mobile | | |
| Desktop | | |

**Review by acaruso-oddball — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jhonnyoddball) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bosawt) [inline .github/PULL_REQUEST_TEMPLATE.md:24]: I think we should probably have different templates for the different repos (even if they're mostly the same), and I'd say the `Screenshots` section is proof of that. Makes sense for frontend, but not really for backend

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bosawt) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: These bullet points should probably be collapsed to a single line, if possible. Actually, I wonder if none of these sections should have a description, and rather we should have a separate document describe the intention for each section.  

The way I currently use our PR template, is I go through each section and delete the comments within, something I've had to do hundreds of time. It would be nice if the actual PR template was set up to expedite PR creation, and other documentation used to train around the process (if we have a separate document then I think all these bullet points are probably relevant)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bosawt) [inline .github/PULL_REQUEST_TEMPLATE.md:43]: I might consider this 'optional', with the 'required' expectations with PR review being something as an organization we generally train for. For example, any PR reviewer should reproduce the 'acceptance criteria' steps, and do a general code review, and probably for many PRs that is sufficient. In the case for those PRs I don't think 'Required Feedback' would need to be filled out

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeniquette) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: This is asking, what tests have you completed as part of the current change you are making. Does that make sense? If not, happy to discuss more.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeniquette) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Yes we will have a slightly different template per repo. I however am not sure showing a screenshot of changes even for vets-api is out of the realm of possibility? Some of the issues we are seeing are that backend changes are being made without ensuring the frontend wasn't changed in a way that causes an issue. I would agree, most backend changes shouldnt have an impact on how the frontend loads things, but more than zero times per month we do.

Thoughts?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeniquette) [reply .github/PULL_REQUEST_TEMPLATE.md:30]: I would defer to the engineer in deciding which of those two options is best for your change. The goal is to ensure the person making the change truly understands the impact of the changes they are making, but also alert any approvers of things that they should be looking out for that may not be obvious based on the code thats being modified.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeniquette) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: Ill mark as optional

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeniquette) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: tracking what your asking. I'd like to keep the template more verbose for the time being and if we can get enough participation then lets work on auto deleting

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (joeniquette) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: For the backend PRs Im going to make screenshots optional

**Comment by joeniquette** [comment]
@jhonnyoddball we good to merge?

**Comment by jhonnyoddball** [comment]
> @jhonnyoddball we good to merge?

I asked the CoP group today in our meeting to take a look at it. If no more comments or reviews until tomorrow, you are good to merge.

**Review by pjhill — APPROVED** [review]
I proposed some changes in a PR of your PR [here](https://github.com/department-of-veterans-affairs/vets-website/pull/22829). I may make some additional proposed changes. I think the current state is a big improvement at any rate. Let's merge it and make additional changes in subsequent PRs.

**Review by powellkerry — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

**Pull request merged**
